LS-27699 Template Vars bug in Lightstep Grafana Metrics Data Source plugin

adil-sultan opened the pull request:
Fixing the bug where 'View What Changed in Lightstep' doesn't pass the value of the templated variable from Grafana.

Addressing this by adding a replace on scoped vars for the query field.

adil-sultan committed:
LS-27699 Template Vars bug in Lightstep

Fixing the bug where View What Changed in Lightstep doesn't pass the value of the templated variable from Grafana.

Addressing this by adding a replace.

smithclay commented:
One small thing: a unit test for the `async query()` function with a test case that covers the string replacement in `datasource.test.js` would be helpful.

adil-sultan committed:
Adding unit test and refactor'd the query method

Refactored the async query method to future proof testing ability.

Also added a unit test for the query method

adil-sultan committed:
addressing linting errors

adil-sultan committed:
more linting errors

adil-sultan committed:
Addressing build and linting errors

removed the public method definition and addressed a type error to correct linting and build error

smithclay commented on src/datasource.ts:
```diff
@@ -155,7 +144,7 @@ export class DataSource extends DataSourceApi<LightstepQuery, LightstepDataSourc
           }
         });
-
+        console.log(field);
```
nit: remove console.log

adil-sultan committed:
removing the stray console.log
